device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/BatchMessages.java

```diff
@@ -0,0 +1,30 @@
+package com.microsoft.azure.sdk.iot.device;
+
+import java.util.List;
+
+public class BatchMessages extends Message
+{
+    /**
+     * List of nested messages.
+     */
+    private List<Message> nestedMessages;
+
+    /**
+     * Creates a batch message
+     * This constructor is internal and can only be called within the com.microsoft.azure.sdk.iot.device package.
+     * @param messages
```

Comment:
"@ param messages" could use a description. Something like "The messages that will be sent as a batch"